src/index.js

```diff
@@ -16,11 +17,33 @@
 import makeHardener from '@agoric/make-hardener';
 import buildTable from './buildTable';
 
-// this use of 'global' is why Harden is a "resource module", whereas
-// MakeHardener is "pure".
-const initialRoots = buildTable((1, eval)('this')); // eslint-disable-line no-eval
-// console.log('initialRoots are', initialRoots);
+// Hunt down our globals.
+const g = typeof globalThis === 'undefined' ? global : globalThis;
```

Comment:
I doubt this works across platforms. If `global` is not defined, I think you'll still need to fallback to an indirect eval of `'this'`. What about a pattern like:

```js
const evalAlias = eval;
... evalAlias('this') ...
```

? Does rollup barf on this?

Reply:
Try this form, used everywhere by test262.

```js
var global = (Function("return this"))();
```

Reply:
I'm not sure if it matters, but this won't work under SES. It relies on the made function being sloppy, and in SES the made function is strict.

Reply:
SES provides its own harden, so no need to access the global!